drivers/usb/gadget/function/f_scm.c

```diff
@@ -0,0 +1,567 @@
+// SPDX-License-Identifier: GPL-2.0+
+/*
+ * f_scm.c -- USB Socket Control Module (SCM) function driver
+ *
+ * Copyright (C) 2018-2019 Xaptum, Inc.
+ */
+
+#include <linux/kernel.h>
+#include <linux/module.h>
+#include <linux/device.h>
+#include <linux/usb/composite.h>
+#include "u_scm.h"
+
+/*
+ * This function is a "Socket Control Module" (SCM) link. The purpose of
+ * SCM is to allow a device to manage and communicate over socekts on its
+ * host. SCMs data transfer model uses bulk endpoints (short packet terminated)
+ * to send and recieve data from the hosts sockets and interrupt endpoints to
+ * communcate socket management data (open, connect, ect).
+ */
+
+#define MAX_INT_PACKET_SIZE 	64
+#define SCM_INTERFACE_SUBCLASS	0xab
+#define SCM_STATUS_INTERVAL_MS	32
+
+/**
+ * Usb function structure definition
+ */
+struct f_scm {
+	struct usb_function     function;
```

Comment:
The indentations in this file are inconsistent. Sometimes tabs, sometimes spaces. For kernel code, it should be all tabs.

I suggest configuring your editor to display a small symbol for each "tab".  Then it's easy to spot inconsistencies.